File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        count_hashmap ={}
        freq_array = [[] for i in range(len(nums)+1)]

        for number in nums:
           count_hashmap[number]= 1 + count_hashmap.get(number,0)
        
        for num,freq in count_hashmap.items():
            freq_array[freq].append(num)
        
        result = []
        for x in range(len(freq_array)-1, 0,-1):
            for num in freq_array[x]:
                result.append(num)
                if len(result) == k:
                   return result
    

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
         set<int> hashset;

         for (int num: nums){
              if (hashset.find(num)==hashset.end()){
                hashset.insert(num);
                } else{
                    return true;
                }    
         }
         return false;
    }
};
